File: pathfinder-api/src/main/java/de/cubbossa/pathapi/storage/cache/NodeTypeCache.java
package de.cubbossa.pathapi.storage.cache;

import de.cubbossa.pathapi.node.Node;
import de.cubbossa.pathapi.node.NodeType;

import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface NodeTypeCache extends StorageCache<UUID> {

  <N extends Node> Optional<NodeType<N>> getType(UUID uuid);

  CacheMap<UUID, NodeType<?>> getTypes(Collection<UUID> uuids);

  void write(UUID uuid, NodeType<?> type);
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/storage/cache/NodeTypeCache.kt
package de.cubbossa.pathfinder.storage.cache

import de.cubbossa.pathfinder.node.Node
import de.cubbossa.pathfinder.node.NodeType
import java.util.*

interface NodeTypeCache : StorageCache<UUID> {
    fun <N : Node> getType(uuid: UUID): NodeType<N>?

    fun getTypes(uuids: Collection<UUID>): StorageCache.CacheMap<UUID, NodeType<*>>

    fun write(uuid: UUID, type: NodeType<*>)
}
